.travis.yml

```diff
@@ -11,8 +11,10 @@ env:
 - PURPOSE=unittest
 - PURPOSE=compile-with-cmake
 - PURPOSE=compile-with-bazel
-#- PURPOSE=compile USE_MESALINK=yes
-#- PURPOSE=unittest USE_MESALINK=yes
+- PURPOSE=compile USE_MESALINK=yes
+- PURPOSE=unittest USE_MESALINK=yes
+- PURPOSE=compile-with-cmake USE_MESALINK=yes
+- PURPOSE=compile-with-bazel USE_MESALINK=yes
```

Comment:
USE_MESALINK option seems not working in cmake and bazel. It is still using openssl.

Reply:
I haven't update the build scripts for cmake and bazel. Let me implement the changes today and push to this PR. Let me know if you think a seperate PR for build scripts would be better.

Reply:
There are many compilation options, such as with-glog, with-thrift. Just compiling with mesalink in travis seems not a general way to test all options. I think it may be a better way to add one specific PURPOSE called compile-with-all-options and we can turn on all the options here, not just mesalink.

Reply:
I have the same feeling; orthogonal build options bloat CI tasks. 

I'm not familiar with the options other than USE_MESALINK; and it seems irrelevant to this PR which only adddresses a previous MesaLink build issue. 

So maybe we merge this PR first; and create a new issue/PR for the proposed compile-with-all-options PURPOSE.